bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -144,19 +144,26 @@ fn build_thumbnail_info(
             let thumbnail_data =
                 fs::read(thumbnail_url).map_err(|_| RoomError::InvalidThumbnailData)?;
 
-            let base_thumbnail_info = BaseThumbnailInfo::try_from(&thumbnail_info)
-                .map_err(|_| RoomError::InvalidAttachmentData)?;
+            let height = thumbnail_info
+                .height
+                .and_then(|u| UInt::try_from(u).ok())
+                .ok_or(RoomError::InvalidAttachmentData)?;
+            let width = thumbnail_info
+                .width
+                .and_then(|u| UInt::try_from(u).ok())
+                .ok_or(RoomError::InvalidAttachmentData)?;
+            let size = thumbnail_info
+                .width
+                .and_then(|u| UInt::try_from(u).ok())
+                .ok_or(RoomError::InvalidAttachmentData)?;
```

Comment:
Since we're failing if any's missing, maybe the FFI's `ThumbnailInfo` could also have non-optional fields for height/width/size? (also the last one is wrong, `let size = thumbnail_info.width`)